homeassistant/components/nasweb/switch.py

```diff
@@ -0,0 +1,95 @@
+"""Platform for NASweb output."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from webio_api import Output as NASwebOutput
+from webio_api.const import KEY_OUTPUTS
+
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import DiscoveryInfoType
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN, OUTPUT_TRANSLATION_KEY
+from .coordinator import NASwebCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Set up switch platform."""
+    # if not isinstance(config, ConfigEntry):
+    #     raise ConfigEntryError("Config is not instance of ConfigEntry")
+    # config_entry: ConfigEntry = config
+    coordinator = hass.data[DOMAIN][config.entry_id]
+    nasweb_outputs = coordinator.data[KEY_OUTPUTS]
+    entities: list[Output] = []
+    for out in nasweb_outputs:
+        if not isinstance(out, NASwebOutput):
+            _LOGGER.error("Cannot create Output entity without NASwebOutput")
+            continue
+        new_output = Output(coordinator, out)
+        entities.append(new_output)
+    async_add_entities(entities)
+
+
+class Output(CoordinatorEntity, SwitchEntity):
+    """Entity representing NASweb Output."""
```

Comment:
Give this a more descriptive class name, Also if it is a coordinator it needs its own file.